squad/ci/models.py

```diff
@@ -68,60 +68,58 @@ def fetch(self, test_job):
             self.really_fetch(test_job)
 
     def really_fetch(self, test_job):
-        implementation = self.get_implementation()
-
         test_job.last_fetch_attempt = timezone.now()
-        test_job.save()
```

Comment:
the intention of saving here is to ensure that if the fetching process crashes with an exception, the date of the last fetch attempts is still recorded. if you only save after the actual fetching, that date will never be updated for jobs that cause a crash

Reply:
The problem is that there might be a few reasons for not an exception here. One of them is that more than one worker picket the fetching task. When the first one finishes it saves actual result to database (for example incomplete job with infra error). When second worker picks the task it overwrites this error with squad internal error saying 'sorry, there already is a test job with this ID'. I asked to simply ignore this exception as it's not very useful from user point of view.

Reply:
We went through the code and it looks like the timestamp will be saved in squad.ci.tasks.fetch (as it is now). So I'm OK to merge this.